### PR TITLE
Re-provision users_view

### PR DESCRIPTION
**Pull request by sureshc**
https://codedotorg.atlassian.net/browse/INF-104

The `users_view` view was provisioned with a raw SQL DDL command embedded in a Rails migration on the reporting database because Rails / ActiveRecord does not support provisioning of Views.  There are 3 problems with this:

1. New environments do not have this View because Rails / ActiveRecord does not include it in schema.rb
2. Anytime we create or rebuild a production database replica, we have to remember to manually create this view, because it does not exist on the production database and does not automatically replicate over to the replica.
3. If/when we migrate to Aurora (which does not permit writing to a replica), we'll need to have this view provisioned on the primary database.

This migration deletes the manually created View from the reporting database and then uses the `Scenic` gem to provision it via Rails.

### CODE REVIEW

**Comment by sureshc** [comment]
@madelynkasula Has Teacher Dashboard moved from Pegasus to Dashboard?  This Pull Request is failing in a [Pegasus test](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/test/test_section_api_helpers.rb) that creates a fake / stubbed dashboard database to test things like section APIs.  I could update this fake dashboard to properly support / stub Scenic and the `create_view` method, but I’m wondering if this Pegasus test can be removed in favor of [existing dashboard tests](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/controllers/api/v1/sections_controller_test.rb).

**Comment by maddiedierker** [comment]
ah, yeah, we should be able to remove section_api_helpers and its tests now that the old teacher dashboard is gone. i'll double-check that nothing is using its classes and prep a PR to remove it.

[update: PR to remove section_api_helpers: #28073]

File: pegasus/test/fixtures/fake_dashboard.rb [hunk 1]
```diff
@@ -247,6 +247,13 @@ def create_table(name, options)
   end
   ActiveRecord::ConnectionAdapters::Mysql2Adapter.prepend SchemaTableFilter
 
+  # Patch Mysql2Adapter to stub create_view when loading the schema.
```
Comment (sureshc) [inline pegasus/test/fixtures/fake_dashboard.rb:250]: @wjordan Is this an acceptable mechanism for stubbing the Scenic `create_view` method?

**Review by uponthesun — COMMENTED** [review]
Are there any tests on Contact Rollups at all? I couldn't find any (though I did find this old commit with tests that was later reverted... https://github.com/code-dot-org/code-dot-org/commit/5bab80f1556c6752c6055c2ae09f18ba008a00fe#diff-037b6adbe18096d1905043dabc2bca48 )

I'd be in favor of spending a little time adding test coverage, though I'd also be open to PLC taking ownership of that if there's a realistic plan for them to do so. Having coverage on that code would also serve to validate that the newly created view is correct. Would like to hear your thoughts on the best bang for buck at this time.

**Comment by sureshc** [comment]
There are no tests, possibly due to the challenge of implementing factories that create the mix and amount of test data to ensure that all of the batches queries and upserts are functioning correctly.  There is a ["validation" step](https://github.com/code-dot-org/code-dot-org/blob/staging/lib/cdo/contact_rollups_validation.rb) that is run with each execution of the job in production that does a sanity check of the results gathered in a temporary `contact_rollups_daily` table, before applying them to the actual `contact_rollups` table.

**Comment by sureshc** [comment]
@madelynkasula I decided to undo the auto-pluralization of the view name to avoid updating existing Contact Rollups queries that use the name `users_view`.  Any thoughts/concerns?

**Comment by maddiedierker** [comment]
@sureshc no, removing auto-pluralization shouldn't cause any issues as long as the naming is consistent everywhere (and it looks like it is)

**Review by uponthesun — APPROVED** [review]

**Pull request merged**